File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (!map.containsKey(nums[i])) {
                map.put(nums[i], 1);
            }
            else {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int size = nums.size();
        vector<int> out(size, 1);
        int pre = 1, suf = 1;
        for (int i = 0; i < size; ++i) {
            out[i] = pre;
            pre *= nums[i];
        }
        for (int i = size-1; i >= 0; --i) {
            out[i] *= suf;
            suf *= nums[i];
        }
        return out;
    }
};
